test/FastTests/Client/Indexing/Counters/BasicCountersIndexes.cs

```diff
@@ -0,0 +1,741 @@
+                Assert.False(staleness.IsStale);
+
+                terms = store.Maintenance.Send(new GetTermsOperation(indexName, "HeartBeat", null));
+                Assert.Equal(1, terms.Length);
+                Assert.Contains("214.5", terms);
+
+                terms = store.Maintenance.Send(new GetTermsOperation(indexName, "Count", null));
+                Assert.Equal(1, terms.Length);
+                Assert.Contains("10", terms);
+
+                terms = store.Maintenance.Send(new GetTermsOperation(indexName, "Name", null));
+                Assert.Equal(1, terms.Length);
+                Assert.Contains("heartrate", terms);
+
+                //// delete documents
+
+                store.Maintenance.Send(new StopIndexingOperation());
+
+                using (var session = store.OpenSession())
+                {
+                    for (int i = 0; i < 20; i++)
+                        session.Delete($"users/{i}");
+
+                    session.SaveChanges();
+                }
+
+                staleness = store.Maintenance.Send(new GetIndexStalenessOperation(indexName));
+                Assert.True(staleness.IsStale);
+                Assert.Equal(1, staleness.StalenessReasons.Count);
+                Assert.True(staleness.StalenessReasons.Any(x => x.Contains("There are still")));
+
+                store.Maintenance.Send(new StartIndexingOperation());
+
+                WaitForIndexing(store);
+
+                staleness = store.Maintenance.Send(new GetIndexStalenessOperation(indexName));
+                Assert.False(staleness.IsStale);
+
+                terms = store.Maintenance.Send(new GetTermsOperation(indexName, "HeartBeat", null));
+                Assert.Equal(0, terms.Length);
+            }
+        }
+
+        [Fact]
+        public async Task BasicMapReduceIndexWithLoad()
+        {
+            using (var store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    for (int i = 0; i < 10; i++)
+                    {
+                        var address = new Address { City = "NY" };
+                        session.Store(address, $"addresses/{i}");
+
+                        var user = new User { AddressId = address.Id };
+                        session.Store(user, $"users/{i}");
+
+                        session.CountersFor(user).Increment("HeartRate", 180 + i);
+                    }
+
+                    session.SaveChanges();
+                }
+
+                store.Maintenance.Send(new StopIndexingOperation());
+
+                string indexName = "AverageHeartRate";
+
+                var result = store.Maintenance.Send(new PutIndexesOperation(new CountersIndexDefinition
+                {
+                    Name = indexName,
+                    Maps = {
+                    "from counter in counters.Users.HeartRate " +
```

Comment:
How can I index all counters in a collection? `counters.Users` would work?

Reply:
Yes, it will work the same way as for timeseries, but I haven't implemented that yet. Going one feature by one.